fetch_cloud.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+)
+
+func FetchCloudData(lat, lon, apiKey string) (*http.Response, error) {
+	url := fmt.Sprintf("http://my.meteoblue.com/packages/clouds-1h_clouds-day?lat=%s&lon=%s&apikey=%s", lat, lon, apiKey)
+	client := http.Client{Timeout: 10 * time.Second}
+	resp, err := client.Get(url)
+	if err != nil {
+		return nil, err
```

Comment:
You're returning the err here, but in the handlerGetCloud, line 28, you just create ur own error instead, saying that it's a server error. I would say that even if ur not using the err, you should atleast log the error so we know what went wrong instead of just sending the error response...

Reply:
yea that can be done